src/lib/receiver.js

```diff
@@ -186,16 +203,31 @@ function createReceiver (opts) {
     }
 
     // The payment request is extracted from the ilp_header
-    let packet = transfer.data && transfer.data.ilp_header
+    const packet = transfer.data && transfer.data.ilp_header
 
     if (!packet) {
       debug('got notification of transfer with no packet attached')
       return rejectIncomingTransfer(transfer.id, 'no-packet')
     }
 
-    // check if the address contains "~ipr" + our receiver id
-    if (packet.account.indexOf(receiverId) === -1) {
-      debug('got notification of transfer for another receiver: ' + packet.account + ' my id is: ' + receiverId)
+    // check if the address starts with our address
+    if (packet.account.indexOf(getAddress()) !== 0) {
```

Comment:
The reason this looks for the receiverId instead of the address is because right now you can pass the address to the `createRequest` function. That was intended for use cases where you want to have one receiver creating and fulfilling requests on behalf of multiple accounts, like in the ilp-kit, but that isn't currently being used. Right now the `ilp-kit` is making one new receiver per payment, but I'm not sure that's the best way to set this up. If we want to keep that behavior we can take out the ability to specify the account for each request.